app/src/main/java/net/gsantner/markor/activity/MainActivity.java

```diff
@@ -103,41 +105,49 @@ public void onPageSelected(int position) {
         }
 
         _cu.applySpecialLaunchersVisibility(this, _appSettings.isSpecialFileLaunchersEnabled());
+    }
 
+    @Override
+    public void onActivityFirstTimeVisible() {
+        super.onActivityFirstTimeVisible();
         // Switch to tab if specific folder _not_ requested, and not recreating from saved instance
         final int startTab = _appSettings.getAppStartupTab();
-        if (startTab != R.id.nav_notebook && savedInstanceState == null && MarkorContextUtils.getValidIntentDir(getIntent(), null) == null) {
+        if (startTab != R.id.nav_notebook && MarkorContextUtils.getValidIntentDir(getIntent(), null) == null) {
             _viewPager.postDelayed(() -> _viewPager.setCurrentItem(tabIdToPos(startTab)), 100);
         }
     }
 
-
     @Override
-    public void onSaveInstanceState(final Bundle outState) {
+    public void onSaveInstanceState(@NonNull final Bundle outState) {
         super.onSaveInstanceState(outState);
 
         // Save references to fragments
-        final FragmentManager manager = getSupportFragmentManager();
-        manager.putFragment(outState, Integer.toString(R.id.nav_notebook), _notebook);
-        manager.putFragment(outState, Integer.toString(R.id.nav_quicknote), _quicknote);
-        manager.putFragment(outState, Integer.toString(R.id.nav_todo), _todo);
-        manager.putFragment(outState, Integer.toString(R.id.nav_more), _more);
+        try {
+            final FragmentManager manager = getSupportFragmentManager();
+            // Put and get notebook first. Most important for correct operation.
+            manager.putFragment(outState, Integer.toString(R.id.nav_notebook), _notebook);
```

Comment:
Catching case when these are null. Happens when we request permissions on resume